Libraries/LibWeb/CSS/StyleComputer.cpp

```diff
@@ -901,7 +902,10 @@ void StyleComputer::collect_animation_into(DOM::AbstractElement abstract_element
         return;
     }
 
-    auto key = static_cast<i64>(round(output_progress.value() * 100.0 * Animations::KeyframeEffect::AnimationKeyFrameKeyScaleFactor));
+    double progress = output_progress.value() >= NumericLimits<double>::max() / (100.0 * Animations::KeyframeEffect::AnimationKeyFrameKeyScaleFactor)
+        ? NumericLimits<double>::max()
+        : round(output_progress.value() * 100.0 * Animations::KeyframeEffect::AnimationKeyFrameKeyScaleFactor);
+    auto key = progress >= NumericLimits<i64>::max() ? NumericLimits<i64>::max() : static_cast<i64>(progress);
```

Comment:
Before this change, UBSAN would complain when `key` was outside the range of `i64`. This change ensures the value of `key` is clamped to the correct range, but the subtests that triggered this failure still fail. They seem to be off by a factor of 10, for reasons I haven't been able to figure out

Reply:
I think it's possible for `progress` to be negative too, that should similarly be clamped to `NumericLimits<i64>::min()`

Reply:
Altered to clamp to the range of i64 in both directions. I've also added a FIXME, since there are WPT tests that rely on a larger value than we now clamp to (although I couldn't find where/if the relevant range is specced).